### PR TITLE
fix: Fix `asyncLoader` type signature

### LINKED ISSUE #309: Some loader's `asyncLoad` argument type is incorrect
-   `three` version: `0.148.0`
-   `node` version: `18.12.1`
-   `yarn` version: `3.3.1`

### Problem description:

The `url` type of `asyncLoad` in `CubeTextureLoader` differs from `load`.
Originally `string[]` in `load` is correct, but since `loadAsync`'s is `string`, it is impossible to load textures correctly when using `loadAsync`.

https://github.com/three-types/three-ts-types/blob/master/types/three/src/loaders/CubeTextureLoader.d.ts#L15

I have not tested it, but it seems that the `HDRCubeTextureLoader` type definition is also incorrect.

https://github.com/three-types/three-ts-types/blob/master/types/three/examples/jsm/loaders/HDRCubeTextureLoader.d.ts

### Relevant code:

```ts
import * as THREE from "three"
const loader = new THREE.CubeTextureLoader();

const urls = ["right.png", "left.png", "top.png", "bottom.png", "front.png", "back.png"];

// it's correct.
loader.load(urls, (texture) => { /* ... */ });

// it's also correct, but will result in a type error.
const texture = await loader.asyncLoad(urls);
```

### Suggested solution:

I propose to remove the `loadAsync` definition from the base `Loader.d.ts` and define it in each loader.
It is necessary to define `loadAsync` in all loaders that inherit `Loader`, but fortunately, most of them have it defined, so we only need to define the missing part.

### PR DESCRIPTION
**Pull request by shun-shobon**
### Why

resolves #309 

### What

- Remove `loadAsync` definition from `Loader`.
- Add `loadAsync` definition to some loaders.
- Fix type signature to `loadAsync` from `CubeTextureLoader` and `HDRCubeTextureLoader`.

### Checklist

-   [x] Checked the target branch (current goes `master`, next goes `dev`)
-   [x] Added myself to contributors table
-   [x] Ready to be merged

### CODE REVIEW

**Comment by shun-shobon** [comment]
I found where `load` and `loadAsync` have different return types. Should I fix it?
https://github.com/three-types/three-ts-types/blob/3ad9711aa8d61f9a5ea0c1a6a89d66c65fc46d84/types/three/examples/jsm/loaders/VOXLoader.d.ts#L12-L18

**Comment by 0b5vr** [comment]
> I found where `load` and `loadAsync` have different return types. Should I fix it?

Blame: https://github.com/three-types/three-ts-types/blame/3ad9711aa8d61f9a5ea0c1a6a89d66c65fc46d84/types/three/examples/jsm/loaders/VOXLoader.d.ts#L12-L18

We probably forgot to change it when we support `VOXMesh` ? I agree to change the return type to `Promise<Chunk[]>`.

File: .all-contributorsrc [hunk 1]
```diff
@@ -494,6 +494,16 @@
...
         "code"
       ]
+    },
+    {
+      "login": "shun-shobon",
+      "name": "Shuntaro Nishizawa",
+      "avatar_url": "https://avatars.githubusercontent.com/u/37236438?v=4",
+      "profile": "https://shun.technology/",
+      "contributions": [
+        "bug",
```
Comment (0b5vr) [inline .all-contributorsrc:504]: correct! most of the people seems to forget this one 😄

**Review by 0b5vr — APPROVED** [review]

**Comment by 0b5vr** [comment]
Thanks! Approved for now but if you could the `VOXLoader` fix we've discussed it would be awesome.

**Comment by shun-shobon** [comment]
> > I found where `load` and `loadAsync` have different return types. Should I fix it?
> 
> Blame: [`3ad9711`/types/three/examples/jsm/loaders/VOXLoader.d.ts#L12-L18 (blame)](https://github.com/three-types/three-ts-types/blame/3ad9711aa8d61f9a5ea0c1a6a89d66c65fc46d84/types/three/examples/jsm/loaders/VOXLoader.d.ts#L12-L18)
> 
> We probably forgot to change it when we support `VOXMesh` ? I agree to change the return type to `Promise<Chunk[]>`.

OK. I changed the return type 😄 (and changed my profile URL from the old one)

**Comment by 0b5vr** [comment]
Thanks!

**Comment by 0b5vr** [comment]
oof I overlooked the target branch. The changes corresponding to the current Three.js versions should have been directed to `master`, not `dev`. It's gonna be released in the next version (r149) anyway so don't worry

**Comment by shun-shobon** [comment]
I should have directed it to master, I misunderstood that it had to be directed to dev. I'm sorry...

**Pull request merged**